common/src/Utilities/Windows/WinMisc.cpp

```diff
@@ -79,147 +79,75 @@ typedef BOOL(WINAPI *PGPI)(DWORD, DWORD, DWORD, DWORD, PDWORD);
-        return L"GetVersionEx Error!";
-
-    // Call GetNativeSystemInfo if supported or GetSystemInfo otherwise.
-
-    pGNSI = (PGNSI)GetProcAddress(GetModuleHandle(L"kernel32.dll"), "GetNativeSystemInfo");
-    if (NULL != pGNSI)
-        pGNSI(&si);
-    else
-        GetSystemInfo(&si);
-
-    if (!IsWindows8Point1OrGreater())
-        return L"Unsupported Operating System!";
-
-    retval += L"Microsoft ";
-
-    // Test for the specific product.
-
-    if (IsWindows10OrGreater())
-        retval += IsWindowsServer() ? L"Windows Server 2016 " : L"Windows 10 ";
-    else // IsWindows8Point1OrGreater()
-        retval += IsWindowsServer() ? L"Windows Server 2012 R2 " : L"Windows 8.1 ";
-
-    pGPI = (PGPI)GetProcAddress(GetModuleHandle(TEXT("kernel32.dll")), "GetProductInfo");
-    pGPI(osvi.dwMajorVersion, osvi.dwMinorVersion, 0, 0, &dwType);
-
-    switch (dwType)
-    {
-        // Shared between Windows 8.1 and 10.
-        case PRODUCT_PROFESSIONAL:
-            retval += L"Pro";
-            break;
-        case PRODUCT_PROFESSIONAL_N:
-            retval += L"Pro N";
-            break;
-        case PRODUCT_PROFESSIONAL_WMC:
-            retval += L"Pro with Media Center";
-            break;
-        case PRODUCT_ENTERPRISE:
-            retval += L"Enterprise";
-            break;
-        case PRODUCT_ENTERPRISE_N:
-            retval += L"Enterprise N";
-            break;
-        case PRODUCT_SERVER_FOUNDATION:
-            retval += L"Foundation";
-            break;
-        case PRODUCT_STANDARD_SERVER:
-            retval += L"Standard";
-            break;
-        case PRODUCT_STANDARD_SERVER_CORE:
-            retval += L"Standard (core)";
-            break;
-        case PRODUCT_DATACENTER_SERVER:
-            retval += L"Datacenter";
-            break;
-        case PRODUCT_DATACENTER_SERVER_CORE:
-            retval += L"Datacenter (core)";
-            break;
-        // Windows 10 specific.
-        case PRODUCT_ENTERPRISE_S:
-            retval += L"Enterprise 2015 LTSB";
-            break;
-        case PRODUCT_ENTERPRISE_S_N:
-            retval += L"Enterprise 2015 LTSB N";
-            break;
-        case PRODUCT_EDUCATION:
-            retval += L"Education";
-            break;
-        case PRODUCT_CORE:
-            retval += L"Home";
-            break;
-        case PRODUCT_CORE_N:
-            retval += L"Home N";
-            break;
-        case PRODUCT_EDUCATION_N:
-            retval += L"Education N";
-            break;
-    }
-
-    retval += wxsFormat(L" (build %d)", osvi.dwBuildNumber);
-
-    if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64)
-        retval += L", 64-bit";
-    else if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_INTEL)
-        retval += L", 32-bit";
-
-    return retval;
+	wxString retval;
+
+	SYSTEM_INFO si;
+	PGNSI pGNSI;
+
+	memzero(si);
+
+	// Call GetNativeSystemInfo if supported or GetSystemInfo otherwise.
```

Comment:
Note: `GetNativeSystemInfo` is XP onwards so it'll always be used.

Reply:
I removed the entire check if you think that's okay. GetProcAddress shouldn't fail in this case correct?

Reply:
Does `GetNativeSystemInfo(&si);` not work?

Reply:
Oh jeez I got it mixed up